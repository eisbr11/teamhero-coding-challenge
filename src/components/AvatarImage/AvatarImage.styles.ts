import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  imageWrap: {
    width: 50,
    height: 50,
    borderRadius: 50,
    display: 'flex',
    overflow: 'hidden',
    border: (theme) => `2px solid ${theme.colorGrayDark}`,
  },
  image: {
    width: '100%',
    objectFit: 'cover',
  },
});

export default useStyles;
