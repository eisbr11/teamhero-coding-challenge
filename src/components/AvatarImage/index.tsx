import { useTheme } from 'react-jss';

import useStyles from './AvatarImage.styles';

const AvatarImage = ({
  src,
  alt,
}: {
  src: string,
  alt: string,
}) => {
  const theme = useTheme();
  const classes = useStyles(theme);
  return (
    <div className={classes.imageWrap}>
      <img src={src} alt={alt} className={classes.image} />
    </div>
  );
};

export default AvatarImage;
