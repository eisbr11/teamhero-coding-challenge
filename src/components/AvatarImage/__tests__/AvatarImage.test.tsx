import React from 'react';
import { ThemeProvider } from 'react-jss';

import { mount } from 'enzyme';
import teamHeroTheme from 'themes/teamHero.theme';
import AvatarImage from '../index';

describe('dataGrid - Component', () => {
  it('should render correctly', () => {
    const avatarImageComponent = mount(
      <ThemeProvider theme={teamHeroTheme}>
        <AvatarImage src="exampleUrl" alt="ALT-TEXT" />
      </ThemeProvider>,
    );

    expect(avatarImageComponent).toMatchSnapshot();
  });
});
