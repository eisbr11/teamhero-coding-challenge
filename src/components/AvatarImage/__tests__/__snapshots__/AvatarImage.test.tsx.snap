// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataGrid - Component should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colorBlack": "#000000",
      "colorGrayDark": "#666666",
      "colorGrayLight": "#f1f4f9",
      "colorHighlight": "#fef0d8",
      "colorPrimary": "#d51641",
      "colorSecondary": "#a3b415",
      "colorWhite": "#FFFFFF",
    }
  }
>
  <AvatarImage
    alt="ALT-TEXT"
    src="exampleUrl"
  >
    <div
      className="imageWrap-0-2-1 imageWrap-d0-0-2-3"
    >
      <img
        alt="ALT-TEXT"
        className="image-0-2-2"
        src="exampleUrl"
      />
    </div>
  </AvatarImage>
</ThemeProvider>
`;
